libautoscoper/src/gpu/cuda/cutil/cutil_create_tex_obj.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include <cuda.h>
+
+inline cudaTextureObject_t createTexFromArray(cudaArray* arr, cudaTextureReadMode readMode) {
+    cudaResourceDesc resDesc;
+    memset(&resDesc, 0, sizeof(resDesc));
+    resDesc.resType = cudaResourceTypeArray;
+    resDesc.res.array.array = arr;
+
+    cudaTextureDesc texDesc;
+    memset(&texDesc, 0, sizeof(texDesc));
+    texDesc.normalizedCoords = true;
+    texDesc.filterMode = cudaFilterModeLinear;
+    texDesc.addressMode[0] = cudaAddressModeClamp;
+    texDesc.addressMode[1] = cudaAddressModeClamp;
+    texDesc.readMode = readMode;
+
+    cudaTextureObject_t tex = 0;
+    cudaCreateTextureObject(&tex, &resDesc, &texDesc, NULL);
```

Comment:
Should we have some error checking here ? (e.g in case the texture object can not be created)